sycl/source/detail/context_impl.cpp

```diff
@@ -206,6 +225,58 @@ pi_native_handle context_impl::getNative() const {
   return Handle;
 }
 
+void context_impl::attachLifetimeToMemObj(std::shared_ptr<const void> &Resource,
+                                          const SYCLMemObjI *AttachTo) {
+  std::lock_guard<std::mutex> lock(MMemObjLifetimeAttachedResourcesMutex);
+  auto AttachedResourcesIt = MMemObjLifetimeAttachedResources.find(AttachTo);
```

Comment:
Minor.
Not sure if it's a good idea, but making `MMemObjLifetimeAttachedResources` to be `std::multimap` here could help to simplify this code. But probably could complicate `detachMemObjLifetimeResources`.

Reply:
Interesting idea. I think you're right that `std::multimap` would make attachment a lot cleaner, but currently I think there's a performance benefit to having it as the current type, specifically because we are able to swap and evict all associated resources easily. I don't know if it is possible for `std::multimap` too.